### PR TITLE
gh-114490: Add check for Mach-O linkage in Lib/platform.py

### PR DESCRIPTION
**Pull request by aidenfoxivey**
<https://github.com/python/cpython/issues/114490>

* Issue: gh-114490

### CODE REVIEW

**Comment by aidenfoxivey** [comment]
Could be wrong here, but this seems like a minor enough change that no news is needed.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
The change itself LGTM.

I would prefer a news entry though, adding it is easy enough and does show users that something changed here.

**Comment by ronaldoussoren** [comment]
BTW. Normally I'd ask for a test as well, but this function is mostly untested at the moment and adding a test for just the linkage on macOS feels wrong to me.

**Comment by aidenfoxivey** [comment]
> The change itself LGTM.
> 
> I would prefer a news entry though, adding it is easy enough and does show users that something changed here.

Sounds good! I'll set it right up.

**Comment by aidenfoxivey** [comment]
> BTW. Normally I'd ask for a test as well, but this function is mostly untested at the moment and adding a test for just the linkage on macOS feels wrong to me.

Oh, so it would be helpful to add some tests for this function?

**Comment by aidenfoxivey** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/macOS/2024-01-23-11-35-26.gh-issue-114490.FrQOQ0.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add Mach-O linkage support for `platform.architecture()`
```
Comment (ronaldoussoren) [inline Misc/NEWS.d/next/macOS/2024-01-23-11-35-26.gh-issue-114490.FrQOQ0.rst]:
```suggestion
Add Mach-O linkage support for :func`platform.architecture()`
```

This fixes the error in the lint job

**Comment by ronaldoussoren** [comment]
> > BTW. Normally I'd ask for a test as well, but this function is mostly untested at the moment and adding a test for just the linkage on macOS feels wrong to me.
> 
> Oh, so it would be helpful to add some tests for this function?

I'm personally not that bothered by not having tests for this function, but better test coverage can help avoiding regressions in the future.

AFAIK the only tests for this function are here:

https://github.com/python/cpython/blob/441affc9e7f419ef0b68f734505fa2f79fe653c7/Lib/test/test_platform.py#L78-L86

Tests will inherently have platform specifics in them, for other functions in `platform` the tests themselves are cross-platform by mocking functions used the test. For `platform.architecture` just mocking `_syscmd_file` should be good enough. This does require access to a number of platforms to collect data though.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
The linting job found a formatting niggle in the news entry, see the suggested change below.

With that change the PR is good to go.

**Comment by aidenfoxivey** [comment]
I have made the requested changes; please review again

File: Misc/NEWS.d/next/macOS/2024-01-23-11-35-26.gh-issue-114490.FrQOQ0.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Add Mach-O linkage support for :func:`platform.architecture()`
```
Comment (aisk) [inline Misc/NEWS.d/next/macOS/2024-01-23-11-35-26.gh-issue-114490.FrQOQ0.rst]:
```suggestion
Add Mach-O linkage support for :func:`platform.architecture()`.
```

Comment (aidenfoxivey) [reply Misc/NEWS.d/next/macOS/2024-01-23-11-35-26.gh-issue-114490.FrQOQ0.rst]: Awesome, thank you. :)

**Review by ronaldoussoren — APPROVED** [review]
LGTM

**Comment by ronaldoussoren** [comment]
@aidenfoxivey, thanks for your PR!

**Pull request merged**